types/defines/email.d.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * A email message that is sent to a consumer Worker.
+ */
+interface EmailMessage<Body = unknown> {
+  /**
+   * A envelope from.
+   */
+  readonly from: string;
+  /**
+   * A envelope to.
+   */
+  readonly to: string;
+  /**
+   * A [Headers object](https://developer.mozilla.org/en-US/docs/Web/API/Headers).
+   */
+  readonly headers: Headers;
+  /**
+   * Stream of the email message content.
+   */
+  readonly raw: ReadableStream;
+  /**
+   * Size of the raw email message content.
+   */
+  readonly rawSize: number;
+  /**
+   * Reject email message
```

Comment:
"Reject this email message by returning a permanent SMTP error back to the connecting client including the given reason."